packages/services/f-feature-management/test/featureGetter.test.js

```diff
@@ -0,0 +1,118 @@
+import { getFeature, init, poll } from '../src/featureGetter';
+import fetchMock from "jest-fetch-mock";
+
+fetchMock.enableMocks();
+
+const settings = {
+  scope: 'test-scope',
+  environment: 'test-env',
+  key: 'test-key'
+};
+
+const fetchResponse = {
+  createdAt: '2021-09-10 15:00',
+  features: [
+    {
+      key: 'key1',
+      testVal: 'val1'
+    }
+  ]
+};
+
+const flushPromises = () => new Promise(res => process.nextTick(res))
+
+describe('When calling poll', () => {  
+
+  beforeEach(async () => {
+    fetch.resetMocks();
+    jest.useFakeTimers();
+
+    fetch.mockResponse(JSON.stringify(fetchResponse));
+
+    await poll(settings);    
+  });
+
+  it('Makes an initial call to fetch', async () => {
+    
+    expect(fetch).toHaveBeenCalledTimes(1);
+    expect(fetch).toHaveBeenCalledWith(`https://features.api.justeattakeaway.com/config/v1/${settings.scope}/${settings.environment}-${settings.key}`);
+  });
+
+  it('Starts polling', async () => {
+    expect(setInterval).toHaveBeenCalledTimes(1);
+    expect(setInterval).toHaveBeenCalledWith(expect.any(Function), 30000);
+
+    jest.advanceTimersByTime(29000);
+    expect(fetch).toHaveBeenCalledTimes(1);    
+
+    jest.advanceTimersByTime(1000);
+    expect(fetch).toHaveBeenCalledTimes(2);
+    expect(fetch).toHaveBeenNthCalledWith(2, `https://features.api.justeattakeaway.com/config/v1/${settings.scope}/${settings.environment}-${settings.key}`);
+  });
+
+  it('Honours different host', async () => {
+    const newSettings = { ...settings, host: 'https://test.com', pollInterval: 60000 };
```

Comment:
We generally have a pattern of having Arrange, Act and Assert comments in each test, would you mind adding those please where applicable?

Reply:
Done